File: 2695-find-score-of-an-array-after-marking-all-elements/2695-find-score-of-an-array-after-marking-all-elements.java
class Solution {
    public long findScore(int[] nums) {
        long ans = 0;
        int[][] sorted = new int[nums.length][2];
        boolean[] marked = new boolean[nums.length];
        for(int i=0; i<nums.length; i++){
            sorted[i][0] = nums[i];
            sorted[i][1] = i;
        }
        Arrays.sort(sorted, (a, b)-> a[0]-b[0]);
        for(int i=0; i<nums.length; i++){
            int num = sorted[i][0], index = sorted[i][1];
            if(!marked[index]){
                ans += num;
                marked[index] = true;
                if(index-1 >= 0) marked[index-1] = true;
                if(index+1 < nums.length) marked[index+1] = true;
            }
        }
        return ans;
    }
}

File: 2695-find-score-of-an-array-after-marking-all-elements/2695-find-score-of-an-array-after-marking-all-elements.cpp
class Solution {
public:
    long long findScore(vector<int>& nums) {
        long long ans = 0;
        vector<pair<int, int>> sorted(nums.size());
        for(int i=0; i<nums.size(); i++) sorted[i] = make_pair(nums[i], i);
        sort(sorted.begin(), sorted.end());
        vector<bool> marked(nums.size(), false);
        for(int i=0; i<nums.size(); i++){
            int num = sorted[i].first, index = sorted[i].second;
            if(!marked[index]){
                ans += num;
                marked[index] = true;
                if(index-1 >= 0) marked[index-1] = true;
                if(index+1 < nums.size()) marked[index+1] = true;
            }
        }
        return ans;
    }
};